optaweb-employee-rostering-backend/src/main/java/org/optaweb/employeerostering/domain/rotation/TimeBucket.java

```diff
@@ -0,0 +1,210 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaweb.employeerostering.domain.rotation;
+
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.persistence.CollectionTable;
+import javax.persistence.ElementCollection;
+import javax.persistence.Entity;
+import javax.persistence.ForeignKey;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+
+import org.optaweb.employeerostering.domain.common.AbstractPersistable;
+import org.optaweb.employeerostering.domain.shift.Shift;
+import org.optaweb.employeerostering.domain.skill.Skill;
+import org.optaweb.employeerostering.domain.spot.Spot;
+
+@Entity
+public class TimeBucket extends AbstractPersistable {
+    private LocalTime startTime;
+    private LocalTime endTime;
+    
+    @ManyToOne
+    private Spot spot;
+    
+    @OneToMany
+    private Set<Skill> additionalSkillSet;
+    
+    @ElementCollection
+    @CollectionTable(name = "repeat_on_day_set",
+        joinColumns = {
+        @JoinColumn(name = "day_id",
+            referencedColumnName = "id",
+            foreignKey=@ForeignKey(name="DAY_FK",
+            foreignKeyDefinition = "FOREIGN KEY (day_id) references public.time_bucket (id)" +
+            " ON UPDATE NO ACTION ON DELETE CASCADE"))
+        }
+    )
+    private Set<DayOfWeek> repeatOnDaySet;
+    
+    @ElementCollection
+    @CollectionTable(name = "seat_list",
+        joinColumns = {
+        @JoinColumn(name = "seat_id",
+            referencedColumnName = "id",
+            foreignKey=@ForeignKey(name="SEAT_FK",
+            foreignKeyDefinition = "FOREIGN KEY (seat_id) references public.time_bucket (id)" +
+            " ON UPDATE NO ACTION ON DELETE CASCADE"))
+        }
+    )
+    private List<Seat> seatList;
+    
+    public TimeBucket() {}
+    
+    public TimeBucket(Integer tenantId, Spot spot, LocalTime startTime, LocalTime endTime,
+                      Set<Skill> additionalSkillSet,
+                      Set<DayOfWeek> repeatOnDaySet, DayOfWeek startOfWeek, int rotationLength) {
+        super(tenantId);
+        this.spot = spot;
+        this.startTime = startTime;
+        this.endTime = endTime;
+        this.additionalSkillSet = additionalSkillSet;
+        this.repeatOnDaySet = repeatOnDaySet;
+        this.seatList = new ArrayList<>(rotationLength);
+        DayOfWeek rotationDay = startOfWeek;
+        for (int i = 0; i < rotationLength; i++) {
+            if (repeatOnDaySet.contains(rotationDay)) {
+                seatList.add(new Seat(i,null));
+            }
+            rotationDay = rotationDay.plus(1);
+        }
```

Comment:
Please extract into a separate function. Constructors ought to be short and concise.
Also, this could be replaced by a relatively simple stream.